File: wk1/project-alpha/backend/app/schemas/ticket.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List
from app.schemas.tag import Tag

class TicketBase(BaseModel):
    title: str
    description: Optional[str] = None
    status: str = "pending"
    priority: str = "medium"

class TicketCreate(TicketBase):
    tag_ids: Optional[List[int]] = []

class TicketUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    priority: Optional[str] = None
    tag_ids: Optional[List[int]] = None

class Ticket(TicketBase):
    id: int
    created_at: datetime
    updated_at: datetime
    tags: List[Tag] = []

    class Config:
        from_attributes = True

File: wk1/project-alpha/frontend/src/types/ticket.ts
import { Tag } from './tag';

export type TicketStatus = 'open' | 'in_progress' | 'closed';
export type TicketPriority = 'low' | 'medium' | 'high';

export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: TicketStatus;
  priority: TicketPriority;
  tags: Tag[];
  created_at: string;
  updated_at: string;
}

export interface TicketCreate {
  title: string;
  description: string;
  status: TicketStatus;
  priority: TicketPriority;
  tag_ids: number[];
}

export interface TicketUpdate {
  title?: string;
  description?: string;
  status?: TicketStatus;
  priority?: TicketPriority;
  tag_ids?: number[];
}

export interface TicketFilter {
  status?: TicketStatus;
  priority?: TicketPriority;
  tag_id?: number;
}
